src/PackageVersions.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration;
+
+use function array_filter;
+use function array_map;
+use function array_merge;
+use function array_reduce;
+use function assert;
+use function file_get_contents;
+use function is_array;
+use function is_object;
+use function is_string;
+use function json_decode;
+
+/**
+ * @internal
+ */
+final class PackageVersions
+{
+    const COMPOSER_INSTALLED = __DIR__ . '/../../../../composer/installed.json';
```

Comment:
Getting the own package version only works on the root package's files and when the package was installed as dependency there.

Therefore only these paths are checked.